src/main/java/ps/as/TransactionDao.java

```diff
@@ -0,0 +1,117 @@
+package ps.as;
+
+import java.sql.*;
+
+public class TransactionDao {
+
+    private Connection connection;
+
+    public TransactionDao() {
+        try {
+            this.connection = DriverManager.getConnection("jdbc:mysql://localhost:3306/budzet_domowy", "root", "aplauz");
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void addTransaction(Transaction transaction) {
+        String sql = "INSERT INTO transaction (type, description, amount, date) VALUES (?,?,?,?)";
+        PreparedStatement preparedStatement;
+        try {
+            preparedStatement = connection.prepareStatement(sql, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setString(1, String.valueOf(transaction.getType()));
+            preparedStatement.setString(2, transaction.getDescription());
+            preparedStatement.setDouble(3, transaction.getAmount());
+            preparedStatement.setString(4, transaction.getDate());
+            preparedStatement.executeUpdate();
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+               transaction.setId(generatedKeys.getInt(1));
+           }
+            System.out.println("Transakcja dodana do bazy danych.");
```

Comment:
Dotyczy to wszystkich metod w tej klasie. Klasa DAO odpowiada za ukrywanie logiki związanej z bazą danych. Metody mogą zwracać wyniki, ale nie powinno być tutaj wyświetlania (printlnów). Tym zajmuj się w miejscu, gdzie z klasy DAO korzystasz, czyli w BudgetApp